x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/elastic/response/ElasticInferenceServiceAuthorizationResponseEntityTests.java

```diff
@@ -7,58 +7,640 @@
+        var authorizedEndpoints = List.of(createElserAuthorizedEndpoint());
+
+        var inferenceIds = authorizedEndpoints.stream()
+            .map(ElasticInferenceServiceAuthorizationResponseEntity.AuthorizedEndpoint::id)
+            .collect(Collectors.toSet());
+
+        return new EisAuthorizationResponse(
+            EIS_ELSER_RESPONSE,
+            new ElasticInferenceServiceAuthorizationResponseEntity(authorizedEndpoints),
+            List.of(createElserExpectedEndpoint(url)),
+            inferenceIds
+        );
+    }
+
+    private static ElasticInferenceServiceModel createElserExpectedEndpoint(String url) {
+        return new ElasticInferenceServiceSparseEmbeddingsModel(
+            ELSER_V2_ENDPOINT_ID,
+            TaskType.SPARSE_EMBEDDING,
+            ElasticInferenceService.NAME,
+            new ElasticInferenceServiceSparseEmbeddingsServiceSettings(ELSER_V2_MODEL_NAME, null),
+            EmptyTaskSettings.INSTANCE,
+            EmptySecretSettings.INSTANCE,
+            new ElasticInferenceServiceComponents(url),
+            new SentenceBoundaryChunkingSettings(250, 1)
+        );
+    }
+
+    private static ElasticInferenceServiceAuthorizationResponseEntity.AuthorizedEndpoint createElserAuthorizedEndpoint() {
+        return new ElasticInferenceServiceAuthorizationResponseEntity.AuthorizedEndpoint(
+            ELSER_V2_ENDPOINT_ID,
+            ELSER_V2_MODEL_NAME,
+            createTaskTypeObject(EIS_SPARSE_PATH, "sparse_embedding"),
+            "preview",
+            List.of("english"),
+            "2024-05-01",
+            null,
+            new ElasticInferenceServiceAuthorizationResponseEntity.Configuration(
+                null,
+                null,
+                null,
+                Map.of("strategy", "sentence", "max_chunk_size", 250, "sentence_overlap", 1)
+            )
+        );
+    }
+
+    public static ElasticInferenceServiceAuthorizationResponseEntity.TaskTypeObject createTaskTypeObject(
+        String eisTaskType,
+        String elasticsearchTaskType
+    ) {
+        return new ElasticInferenceServiceAuthorizationResponseEntity.TaskTypeObject(eisTaskType, elasticsearchTaskType);
+    }
+
+    public static EisAuthorizationResponse getEisAuthorizationResponseWithMultipleEndpoints(String url) {
+        var authorizedEndpoints = List.of(
+            createRainbowSprinklesAuthorizedEndpoint(),
+            createGpLlmV2ChatCompletionAuthorizedEndpoint(),
+            createGpLlmV2CompletionAuthorizedEndpoint(),
+            createElserAuthorizedEndpoint(),
+            createJinaEmbedAuthorizedEndpoint(),
+            new ElasticInferenceServiceAuthorizationResponseEntity.AuthorizedEndpoint(
+                RERANK_V1_ENDPOINT_ID,
+                RERANK_V1_MODEL_NAME,
+                createTaskTypeObject(EIS_RERANK_PATH, "rerank"),
+                "preview",
+                List.of(),
+                "2024-05-01",
+                null,
+                null
+            )
```

Comment:
Nitpick, but just for consistency, could the rerank creation be extracted to a method like the other endpoint types?